sbtc/tests/integration/validation.rs

```diff
@@ -81,6 +81,115 @@ fn tx_validation_from_mempool() {
+    let amount = 30_000_000;
+
+    // 1. Create and submit a transaction where the lock script has a
+    //    non-minimal push for the deposit.
+    //
+    // We're going to create a P2SH UTXO with a script that has a
+    // non-minimal push of data. Anyone can spend this script immediately.
+    let deposit_data = [0; 44];
+
+    // This is non-standard script because it does not follow the minimal
+    // push rule. We use the OP_PUSHDATA1 when we can use fewer bytes for
+    // the same outcome in the script by removing the OP_PUSHDATA1 opcode
+    // from the script.
+    let script_pubkey = ScriptBuf::builder()
+        .push_opcode(bitcoin::opcodes::all::OP_PUSHDATA1)
+        .push_slice(deposit_data)
+        .push_opcode(bitcoin::opcodes::all::OP_DROP)
+        .push_opcode(bitcoin::opcodes::OP_TRUE)
+        .into_script();
+
+    let mut tx0 = Transaction {
+        version: Version::ONE,
+        lock_time: LockTime::ZERO,
+        input: vec![TxIn {
+            previous_output: utxo.outpoint(),
+            sequence: Sequence::ZERO,
+            script_sig: ScriptBuf::new(),
+            witness: Witness::new(),
+        }],
+        output: vec![
+            TxOut {
+                value: Amount::from_sat(amount),
+                script_pubkey: ScriptBuf::new_p2sh(&script_pubkey.script_hash()),
+            },
+            TxOut {
+                value: utxo.amount() - Amount::from_sat(amount + fee),
+                script_pubkey: depositor.address.script_pubkey(),
+            },
+        ],
+    };
+
+    regtest::p2tr_sign_transaction(&mut tx0, 0, &[utxo], &depositor.keypair);
+    rpc.send_raw_transaction(&tx0).unwrap();
+
+    // 2. Confirm the transaction and try spend it immediately. The
+    //    transaction spending the "deposit" should be rejected.
+    faucet.generate_blocks(1);
+
+    // The Builder::push_slice wants to make sure that the length of the
+    // pushed data is within the limits, hence the conversion into this
+    // PushBytes thing.
+    let locking_script: &PushBytes = script_pubkey.as_bytes().try_into().unwrap();
+    let script_sig = ScriptBuf::builder()
+        .push_slice(locking_script)
+        .into_script();
+
+    let tx1 = Transaction {
+        version: Version::TWO,
+        lock_time: LockTime::ZERO,
+        input: vec![TxIn {
+            previous_output: OutPoint::new(tx0.compute_txid(), 0),
+            sequence: Sequence::ZERO,
+            script_sig,
+            witness: Witness::new(),
+        }],
+        output: vec![TxOut {
+            value: Amount::from_sat(amount - fee),
+            script_pubkey: depositor.address.script_pubkey(),
+        }],
+    };
+    // We just created a transaction that spends the P2SH UTXO where we
+    // spend a deposit that did not adhere to the minimal push rule. When
+    // bicoin-core attempts to validate the transaction it should fail.
+    match rpc.send_raw_transaction(&tx1).unwrap_err() {
```

Comment:
The following is a nicer way to check this imo (no change required, just a suggestion):

```rust
assert!(matches!(
    rpc.send_raw_transaction(&tx1).unwrap_err(),
    BtcRpcError::JsonRpc(JsonRpcError::Rpc(err))
        if err.code == -26 && err.message == "non-mandatory-script-verify-flag (Data push larger than necessary)"
))
```

Reply:
Yeah I like it more aesthetically, but it's harder to debug if bitcoin-core returns a different error response. But I changed it to combine the best of both approaches.